state/executor.go

```diff
@@ -133,19 +134,49 @@ func (e *Executor) ProcessBlock(
 	block *types.Block,
 	blockCreator types.Address,
 ) (*Transition, error) {
+	e.logger.Debug("[BlockchainWrapper.ProcessBlock]",
+		"block number", block.Number(), "block hash", block.Hash(),
+		"parent state root", parentRoot, "block state root", block.Header.StateRoot)
+
 	txn, err := e.BeginTxn(parentRoot, block.Header, blockCreator)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, t := range block.Transactions {
+	var buf bytes.Buffer
+
+	err = block.TxnIterator(func(i int, t *types.Transaction) *types.IterationResult {
 		if t.Gas() > block.Header.GasLimit {
-			continue
+			return &types.IterationResult{ShouldContinue: true}
 		}
 
 		if err = txn.Write(t); err != nil {
```

Comment:
Should we log transaction before `transition.Write`, in case it fails, we can see the details of the tx?

Reply:
Fixed https://github.com/Ethernal-Tech/blade/pull/195/commits/79c0e49d286636fca7bec125db980ace70d842f5